obsplus/structures/fetcher.py

```diff
@@ -536,7 +568,9 @@ def download_event_waveforms(
         # iter events and save to disk
         t1, t2 = time_before_origin, time_after_origin
         for event_id, stream in self.yield_event_waveforms(t1, t2):
-            bank.put_waveforms(stream, name=event_id)
+            # get name, need to make valid for file names on all systems
+            name = event_id.replace("smi:local/", "").replace(":", "-")
```

Comment:
Nice, this "smi:local/" is the only thing I have seen in event_ids, but I am not certain this is the only prefix that you might see? Do you also need to replace any remaining "/" characters, just in case?

Reply:
And, maybe this change in event_id should be handled by `bank.put_waveforms` internally, to avoid repeated code, and make sure this is path-safe in various places?

Reply:
Yes, actually there might be a problem with the way `Fetcher` is interacting with the `WaveBank`, I will need to look more into it.

Reply:
After looking at this more I think its best to remove the download functions; it is only a few lines of code and will allow the user more freedom in picking naming scheme. Otherwise, there is some guesswork involved in intuiting what the user might want, which I would rather avoid.